libs/x-content/src/main/java/org/elasticsearch/common/xcontent/AbstractObjectParser.java

```diff
@@ -222,6 +223,22 @@ public void declareIntOrNull(BiConsumer<Value, Integer> consumer, int nullValue,
                 field, ValueType.INT_OR_NULL);
     }
 
+    /**
+     * Declare a field of type {@code T} parsed from string and converted to {@code T} using provided function.
+     * Throws if the next token is not a string.
+     */
+    public <T> void declareStringField(BiConsumer<Value, T> consumer, Function<String, T> fromStringFunction, ParseField field) {
+        declareField(
+            consumer,
+            p -> {
+                if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
```

Comment:
I *thought* that this wouldn't get called unless it is on a string because of the `ValueType.STRING`.